frontend/src/features/crawl-workflows/workflow-tag-filter.ts

```diff
@@ -0,0 +1,242 @@
+
+          for (const [tag, value] of this.selected) {
+            if (value) {
+              selectedTags.push(tag);
+            }
+          }
+
+          this.dispatchEvent(
+            new CustomEvent<BtrixChangeEvent["detail"]>("btrix-change", {
+              detail: { value: selectedTags.length ? selectedTags : undefined },
+            }),
+          );
+        }}
+      >
+        ${this.tags?.length
+          ? html`<span class="text-primary-500">${msg("Tagged")}</span>
+              ${this.renderTagsInLabel(this.tags)}`
+          : msg("Tags")}
+
+        <div
+          slot="dropdown-header"
+          class=${clsx(this.orgTagsTask.value && tw`border-b`, tw`pb-3`)}
+        >
+          <div class="flex items-center justify-between py-1">
+            <sl-menu-label class="part-[base]:px-4" id="tag-list-label">
+              ${msg("Filter by Tags")}
+            </sl-menu-label>
+            ${this.tags?.length
+              ? html`<sl-button
+                  variant="text"
+                  size="small"
+                  @click=${() => {
+                    this.dispatchEvent(
+                      new CustomEvent<BtrixChangeEvent["detail"]>(
+                        "btrix-change",
+                        {
+                          detail: {
+                            value: undefined,
+                          },
+                        },
+                      ),
+                    );
+                  }}
+                  >${msg("Clear Filter")}</sl-button
+                >`
+              : nothing}
+          </div>
+
+          <div class="px-3">${this.renderSearch()}</div>
+        </div>
+
+        <div slot="dropdown-content" class="contents">
+          ${this.orgTagsTask.render({
+            complete: (tags) => {
+              let options = tags;
+
+              if (tags.length && this.searchString) {
+                options = this.fuse.search(this.searchString);
+              }
+
+              if (options.length) {
+                return this.renderList(options);
+              }
+
+              return html`<div class="p-3 text-neutral-500">
+                ${this.searchString
+                  ? msg("No matching tags found.")
+                  : msg("No tags found.")}
+              </div>`;
+            },
+          })}
+        </div>
+      </btrix-workflow-filter>
+    `;
+  }
+
+  private renderTagsInLabel(tags: string[]) {
+    const more = () => {
+      const remainder = this.localize.number(tags.length - MAX_TAGS_IN_LABEL);
+
+      return html`<span class="ml-2 text-primary-500">
+        ${msg(str`+ ${remainder} more`)}
+      </span>`;
+    };
+    return html`${tags
+      .slice(0, MAX_TAGS_IN_LABEL)
+      .map(
+        (tag, i) =>
+          html`${i > 0 ? html`<span class="opacity-50">, </span>` : ""}<strong
+              class="font-semibold"
+              >${tag}</strong
+            >`,
+      )}${tags.length > MAX_TAGS_IN_LABEL ? more() : nothing}`;
```

Comment:
`Intl.FormatList` (introduced in ES2021, baseline/widely available) would be a good fit for this, maybe something like this?

```suggestion
    const formatter = new Intl.ListFormat(localize.activeLocales, {
      style: "long",
      type: "conjunction",
    });
    return formatter
      .formatToParts(
        tags.length > MAX_TAGS_IN_LABEL
          ? [
              ...tags.slice(0, MAX_TAGS_IN_LABEL),
              msg(
                str`${this.localize.number(tags.length - MAX_TAGS_IN_LABEL)} more`,
              ),
            ]
          : tags,
      )
      .map((part, index, array) =>
        part.type === "literal"
          ? html`<span class="opacity-50">${part.value}</span>`
          : tags.length > MAX_TAGS_IN_LABEL && index === array.length - 1
            ? html`<span class="text-primary-500"> ${part.value} </span>`
            : html`<strong class="font-semibold">${part.value}</strong>`,
      );
```

Reply:
Nice suggestion!